services/repository-managers/src/main/java/org/sagebionetworks/repo/util/jrjc/JRJCHelper.java

```diff
@@ -41,28 +41,28 @@ public static String createRestrictIssue(JiraClient jiraClient, String principal
 	 * @param params a map from field names to field values
 	 * @return
 	 */
-	public static JSONObject createIssue(JiraClient jiraClient, String issueTypeName, String summary, Map<String,String> params) {
-		JSONObject projectInfo = jiraClient.getProjectInfo(JIRA_PROJECT_KEY, issueTypeName);
-		String projectId = (String) projectInfo.get("id");
-		Long issueTypeId = (Long) projectInfo.get("issueTypeId");
-
+	public static CreatedIssue createIssue(JiraClient jiraClient, String issueTypeName, String summary, Map<String,String> params) {
+		ProjectInfo projectInfo = jiraClient.getProjectInfo(JIRA_PROJECT_KEY, issueTypeName);
+		String projectId = projectInfo.getProjectId();
+		Long issueTypeId = projectInfo.getIssueTypeId();
 		Map<String, String> fieldsMap = jiraClient.getFields();
-		JSONObject issue = new JSONObject();
-		JSONObject issueFields = new JSONObject();
-		JSONObject o = new JSONObject();
-		issueFields.put("summary", summary);
-		o = new JSONObject();
-		o.put("id", projectId);
-		issueFields.put("project", o);
-		o = new JSONObject();
-		o.put("id", issueTypeId);
-		issueFields.put("issuetype", o);
+
+		BasicIssue basicIssue = new BasicIssue();
+		basicIssue.setProjectId(projectId);
+		basicIssue.setIssueTypeId(issueTypeId);
+		basicIssue.setSummary(summary);
+		basicIssue.setCustomFields(mapParams(fieldsMap, params));
+
+		CreatedIssue createdIssue = jiraClient.createIssue(basicIssue);
+		return createdIssue;
+	}
+
+	private static Map<String, String> mapParams(Map<String, String>fieldsMap, Map<String, String> params) {
+		Map<String, String> mapped = new HashMap<>();
 		for (String k: params.keySet()) {
 			String fk = fieldsMap.get(k);
```

Comment:
This could be null, we should probably handle this if the JIRA api changes again